### PR TITLE
[frontend] Improve

### LINKED ISSUE #160: Remove unused asset, JSON

### LINKED ISSUE #162: blocking when shop disable
when shop disable them:

- [x] edit/add/delete coupon
- [x] edit/add/delete product
- [x] add new tag

will throw ***500***

### LINKED ISSUE #163: Improve product pages UX
waiting fix:

- [x] checkbox too many space
- [x] not allow no tag
- [x] cancel button wrong name as delete in add product
- [x] lack cancel button in edit product
- [ ] some Not OK Response doesn't throw error message

### PR DESCRIPTION
**Pull request by KatoToyoki**
Resolves 
some of 
#160
#162
#163 

### Changes

some of 
- #160
  - Remove unused asset, JSON
- #162
  - [x] edit/add/delete coupon
  - [x] edit/add/delete product
  - [x] add new tag
- #163 
  - [x] checkbox too many space
  - [x] not allow no tag
  - [x] cancel button wrong name as delete in add product
  - [x] lack cancel button in edit product
  - [ ] some Not OK Response doesn't throw error message
### Test Environment Setup
open 3 terminals
one for `docker-compose up postgres minio `
one for `go run .`
one for `cd frontend` -> `npm i` -> `npm run build`
then open the browser then enter http://localhost:8080/ 

### Additional context

N / A

### CODE REVIEW

**Comment by UGisBusy** [comment]
this will need to fix after #168 merged
( backend will not return 200 when it failed

**Comment by kryr34** [comment]
can #157 #160 be separated into 2 other PR?
and please list detail change about *what some of* #162 #163 #164

**Comment by KatoToyoki** [comment]
> this will need to fix after #168 merged ( backend will not return 200 when it failed

fix where
i add all `patch`, `post`, `delete` 500 detection

**Comment by KatoToyoki** [comment]
> can #157 #160 be separated into 2 other PR? and please list detail change about _what some of_ #162 #163 #164

i have no time tonight i said several times thanks, but i can move issue detail to here
edit : here i use improper word, my bad

**Review by UGisBusy — CHANGES_REQUESTED** [review]
> #164 

u didnt refactor any error message, there're many grammar error
if u wont fix it in this pr, pls dont link it and checked it.

**Comment by KatoToyoki** [comment]
> > #164
> 
> u didnt refactor any error message, there're many grammar error if u wont fix it in this pr, pls dont link it and checked it.

what but i use suggestion code of @kryr34 and the alert is also updated? i ask chatGPT to correct the grammer.
can you point out where is not changed?

**Comment by kryr34** [comment]
> i have no time tonight i said several times thanks, but i can move issue detail to here

if you don't have time to complete your work, there's no need to hastily submit incomplete work

File: frontend/src/pages/search/index.tsx [hunk 1]
```diff
@@ -115,100 +111,77 @@ const toOrderBy = (input: string | null) => {
...
 
-  if (
-    data.maxPrice !== null &&
-    data.minPrice !== null &&
-    parseInt(data.maxPrice.toString()) < parseInt(data.minPrice.toString())
-  ) {
-    alert("min price can't bigger than max value");
-    return false;
+  if (isMinBiggerThanMax(data.minPrice, data.maxPrice)) {
+    return AlertWithReturn("Min price can't be bigger than max values.");
```
Comment (UGisBusy) [inline frontend/src/pages/search/index.tsx]:
```suggestion
    return AlertWithReturn("Min price can't be greater than max price.");
```

File: frontend/src/pages/search/index.tsx [hunk 2]
```diff
@@ -115,100 +111,77 @@ const toOrderBy = (input: string | null) => {
...
+    max !== null &&
+    Number(min.toString()) > Number(max.toString());
+
+  const isNegative = (value: number | string | null) =>
+    value !== null && parseInt(value.toString()) < 0;
+
+  if (isEntered(data.maxPrice)) {
+    if (!isValidNumber(data.maxPrice))
+      return AlertWithReturn('Please enter numbers for the max price.');
+    if (isNegative(data.maxPrice)) return AlertWithReturn("Max price can't be negative numbers.");
```
Comment (UGisBusy) [inline frontend/src/pages/search/index.tsx]:
```suggestion
    if (isNegative(data.maxPrice)) return AlertWithReturn("Max price can't be negative.");
```

File: frontend/src/pages/search/index.tsx [hunk 3]
```diff
@@ -115,100 +111,77 @@ const toOrderBy = (input: string | null) => {
...
-    alert("min price can't negative numbers");
-    return false;
-  }
+  const isFloat = (value: number | string | null) =>
+    !isInteger(value) && value !== null && value.toString() !== '';
 
-  if (data.maxPrice !== null && parseInt(data.maxPrice.toString()) < 0) {
-    alert("max price can't negative numbers");
+  if ([data.maxPrice, data.minPrice, data.maxStock, data.minStock].some(isFloat)) {
+    alert("Can't enter float numbers!");
```
Comment (UGisBusy) [inline frontend/src/pages/search/index.tsx]:
```suggestion
    alert("please enter integers for min/max price and min/max stock.");
```

**Comment by KatoToyoki** [comment]
> > i have no time tonight i said several times thanks, but i can move issue detail to here
> 
> if you don't have time to complete your work, there's no need to hastily submit incomplete work

the work i mentioned here are all complete so i open the pr to let you check. 
when i fix products, @UGisBusy said i can also modify coupons.
or where do you think it's hastily submit, please point out.

File: frontend/src/pages/search/index.tsx [hunk 4]
```diff
@@ -115,100 +111,77 @@ const toOrderBy = (input: string | null) => {
...
+  if (isMinBiggerThanMax(data.minStock, data.maxStock)) {
+    return AlertWithReturn("Min stock can't be bigger than max value.");
   }
 
-  if (isEmpty(data.maxPrice) != isEmpty(data.minPrice)) {
-    alert('max and min price should both have values or both have no values at the same time');
-    return false;
+  if (isEntered(data.maxPrice) !== isEntered(data.minPrice)) {
+    return AlertWithReturn(
+      'Max and min price should both have valid values or both have no values at the same time.',
```
Comment (UGisBusy) [inline frontend/src/pages/search/index.tsx]:
```suggestion
      'please enter max/min price when the other one is presented.',
```

**Comment by UGisBusy** [comment]
can u split #157 into another pr?
that one might cause bugs and need to be tested cautiously while others just some minor fix.

**Comment by kryr34** [comment]
my bad

**Review by x06lan — APPROVED** [review]

**Review by Error0229 — APPROVED** [review]
LGTM

**Review by UGisBusy — APPROVED** [review]
lgtm

**Review by UGisBusy — APPROVED** [review]
lgtm

**Pull request merged**